src/executor/insert_executor.cpp

```diff
@@ -299,9 +328,19 @@ bool InsertExecutor::DExecute() {
                                    current_txn);
       }
     }
+    // Unlock the table
+    bool unlock_success = lm->UnlockShared(table_oid);
+    if (!unlock_success) {
+      LOG_TRACE("Cannot unlock the table, abort!");
+    }
     done_ = true;
     return true;
   }
+  // Unlock the table
+  bool unlock_success = lm->UnlockShared(table_oid);
```

Comment:
If negative case for unlock is quite same, maybe we could write some wrapper to do this instead of write similar logic.